File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/main/java/com/vaadin/flow/component/combobox/test/template/WrapperTemplate.java
/*
 * Copyright (C) 2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See {@literal <https://vaadin.com/commercial-license-and-service-terms>}  for the full
 * license.
 */
package com.vaadin.flow.component.combobox.test.template;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.HtmlImport;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.polymertemplate.Id;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.templatemodel.TemplateModel;

@Tag("wrapper-template")
@HtmlImport("src/wrapper-template.html")
@JsModule("./src/wrapper-template.js")
public class WrapperTemplate extends PolymerTemplate<TemplateModel> {

    @Id
    ComboBoxInATemplate comboBoxInATemplate;

    @Id
    ComboBoxInATemplate2 comboBoxInATemplate2;
}

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/frontend/src/wrapper-template.js
/*
 * Copyright (C) 2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See {@literal <https://vaadin.com/commercial-license-and-service-terms>}  for the full
 * license.
 */

import {PolymerElement} from '@polymer/polymer/polymer-element.js';
import {html} from '@polymer/polymer/lib/utils/html-tag.js';

class WrapperTemplate extends PolymerElement {
  static get template() {
    return html`
        <combo-box-in-a-template id="comboBoxInATemplate"></combo-box-in-a-template>
        <combo-box-in-a-template2 id="comboBoxInATemplate2"></combo-box-in-a-template2>
`;
  }

  static get is() {
      return 'wrapper-template'
  }
}
customElements.define(WrapperTemplate.is, WrapperTemplate);
